proposals/001-operator-design.md

```diff
@@ -0,0 +1,818 @@
+      sectionName: my-listener
+      brokers:
+        - advertisedBrokerIds: [ 1,2,3 ]
+          brokerIdentityMapping: identity
+      hostnames:
+        - my-cluster-%.svc.cluster.local
+      rules:
+        - name: my-rule
+          filters: [ ]
+          backendRefs:
+            - kind: Service
+              namespace: kafka
+              name: my-cluster
+              port: 9093
+```
+
+Because the hostname is suffixed with `svc.cluster.local` the operator will infer that we want to expose this hostname
+in-cluster. The operator will create:
+
+1. A ClusterIP Service named `my-cluster-bootstrap` mapping port 9092 to all the Proxy instances
+2. A ClusterIP Service named `my-cluster-broker-1` mapping port 9092 to all the Proxy instances
+3. A ClusterIP Service named `my-cluster-broker-2` mapping port 9092 to all the Proxy instances
+4. A ClusterIP Service named `my-cluster-broker-3` mapping port 9092 to all the Proxy instances
+
+The certificate secret will be mounted into the pod at a known location
+like `/opt/kroxylicious/secrets/tls/${listenerName}/${certSecretName}/`
+
+The proxy will be configured with a single VirtualCluster with SNI exposition (bootstrap ==
+my-cluster-bootstrap.my-namespace.svc.cluster.local, brokerPattern = my-cluster-broker-$(nodeId)
+.my-namespace.svc.cluster.local).
+
+The VirtualCluster will have the certificate wired in and know how to select the certificate as discussed here.
+
+The virtual cluster target cluster bootstrap will be configured as `my-cluster.kafka:9093`
+
+From there, the client can bootstrap with `my-cluster-bootstrap.my-namespace.svc.cluster.local:9092` within the cluster.
+
+### 2. Off-Cluster access
+
+```yaml
+---
+kind: KafkaGateway
+apiVersion: kroxylicious.io/v1alpha1
+metadata:
+  name: simple
+  namespace: my-namespace
+spec:
+  listeners:
+    - name: my-listener
+      port: 9092 # required
+      protocol: kroxylicious.io/kafka-tls
+      tls:
+        certificateRefs:
+          - # user supplied secret containing tls.crt and tls.key
+            kind: Secret
+            name: my-cert
+            namespace: my-namespace
+---
+kind: KafkaRoute
+apiVersion: kroxylicious.io/v1alpha1
+metadata:
+  name: my-route
+  namespace: my-namespace
+spec:
+  parentRefs:
+    - group: kroxylicious.io
+      kind: KafkaGateway
+      name: simple
+      sectionName: my-listener
+      brokers:
+        - advertisedBrokerIds: [ 1,2,3 ]
+          brokerIdentityMapping: identity
+      hostnames:
+        - my-cluster-%.example.com
+      rules:
+        - name: my-rule
+          filters: [ ]
+          backendRefs:
+            - kind: Service
+              namespace: kafka
+              name: my-cluster
+              port: 9093
+```
+
+Because the hostname is not suffixed with `svc.cluster.local` the operator will infer that we want to expose this
+hostname off-cluster.
+
+The operator will create a LoadBalancer Service named `my-route` mapping port 9092 to all the Proxy instances
```

Comment:
Sometimes you want to make hints to the loadbalancer to control its behaviour.  This is done via the Service's [annotations](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/). 

We've got the `infrastructure:` field at the Gateway, but that will apply to all the resources created by the operator.   AFAICS, we have no way to target an annotation at a particular resource.